Godeps/_workspace/src/github.com/Clever/gearadmin/Makefile

```diff
@@ -0,0 +1,20 @@
+SHELL := /bin/bash
+PKG = github.com/Clever/gearadmin
+SUBPKGS =
```

Comment:
The only bloated commits are when you change dependencies (rare) and the
initial switch to godep (happens once) so I don't see much of a cost there.

I've seen this argument play out in a bunch of open source repos (I PR'd
the original godep usage in drone in the way you describe) but in then end
trying to vendor without really vendoring ends up complicating build
processes and local development since you have a bunch of extra unknown
state (i.e. the content of your empty vendor directory). I've come around
pretty strongly to vendoring the way it's done in the vast majority of
cases, i.e. committing deps



On Jan 5, 2015, at 10:11 PM, Alex Zylman <notifications@github.com> wrote:

In Godeps/_workspace/src/github.com/Clever/gearadmin/Makefile
<https://github.com/Clever/gearman-load-logger/pull/1#discussion-diff-22510100>
:

> @@ -0,0 +1,20 @@
> +SHELL := /bin/bash
> +PKG = github.com/Clever/gearadmin
> +SUBPKGS =

In the spirit if "Don't Trust The Textbook", I'm going to ask "why?" The
way I see it, the biggest benefit of godeps is locking down to a specific
commit, which is unrelated to whether or not you check that code into your
repo.

The upside I see to checking it in: faster build times + test times, since
you don't need to redownload (you could also get some of this benefit in
other ways, e.g. by having Drone cache the _workspace directory)
The downside: bloated repositories + commits

—
Reply to this email directly or view it on GitHub
<https://github.com/Clever/gearman-load-logger/pull/1/files#r22510100>.